sway-error/src/error.rs

```diff
@@ -3112,6 +3122,32 @@ impl ToDiagnostic for CompileError {
                     help
                 },
             },
+            ABIDuplicateName { span, other_span: other, is_attribute } => Diagnostic {
+                reason: Some(Reason::new(code(1), "Duplicated name found for renamed ABI type.".into())),
```

Comment:
According to the guidelines, `Reason`s shouldn't end up in punctuation (we have messages that violates that, I know...)
```suggestion
                reason: Some(Reason::new(code(1), "Duplicated name found for renamed ABI type".into())),
```